### PR TITLE
24454 - EFT TDI17 Processing updates

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/24454

*Description of changes:*
- configurable patterns short name parsing for EFT/EFT WIRE
- configuration validation on eft-reconcilliation
- generalized unknown pattern short name generation
- clean up eft_file columns that are no longer relevant
- restructure tests
- remove PAD check

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/migrations/versions/2024_11_19_8bd139bbb602_.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+revision = '8bd139bbb602'
+down_revision = '474917a13bd4'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table('eft_files', schema=None) as batch_op:
+        batch_op.drop_column('total_deposit_cents')
+        batch_op.drop_column('number_of_details')
```
Comment (ochiu) [inline pay-api/migrations/versions/2024_11_19_8bd139bbb602_.py:20]: Removed so there is no confusion. These are not accurate as they are extracted directly from the TDI17 which include extra transactions that are not EFT.

We filter by location_id during processing to only get EFT transactions, which the total may not match the file total.

File: pay-queue/tests/integration/test_eft_reconciliation.py [hunk 2]
```diff
@@ -737,7 +728,7 @@ def create_test_data():
     return payment_account, eft_short_name, invoice
 
 
-def generate_basic_tdi17_file(file_name: str):
+def generate_basic_tdi17_file(app, file_name: str):
```
Comment (ochiu) [inline pay-queue/tests/integration/test_eft_reconciliation.py:731]: Updated to test all configured patterns

File: pay-queue/tests/integration/test_eft_reconciliation.py [hunk 3]
```diff
@@ -310,11 +304,11 @@ def test_eft_tdi17_fail_transactions(session, app, client, mocker):
     assert "Invalid transaction deposit amount CAD." == call_args[0].error_messages[0]["error"]
 
 
-def test_eft_tdi17_basic_process(session, app, client):
-    """Test EFT Reconciliations worker is able to create basic EFT processing records."""
+def test_eft_tdi17_all_patterns_process(session, app, client):
```
Comment (ochiu) [inline pay-queue/tests/integration/test_eft_reconciliation.py:307]: refactored to test all configured patterns with short names and generated short names with no defined pattern

File: pay-queue/src/pay_queue/services/eft/eft_record.py [hunk 4]
```diff
@@ -113,20 +113,17 @@ def parse_transaction_description(self):
...
+        if matching_pattern := self.find_matching_pattern(self.eft_wire_patterns, self.transaction_description):
             self.short_name_type = EFTShortnameType.WIRE.value
-            self.transaction_description = self.transaction_description[len(self.WIRE_DESCRIPTION_PATTERN) :].strip()
+            self.transaction_description = self.transaction_description[len(matching_pattern) :].strip()
             return
 
-        # Check if this a PAD or EFT Transaction
-        if self.transaction_description.startswith(
-            self.EFT_DESCRIPTION_PATTERN
-        ) and not self.transaction_description.startswith(self.PAD_DESCRIPTION_PATTERN):
```
Comment (ochiu) [inline pay-queue/src/pay_queue/services/eft/eft_record.py:130]: PAD check no longer needed when using location id filter when processing.

File: pay-queue/src/pay_queue/services/eft/eft_reconciliation.py [hunk 5]
```diff
@@ -61,6 +61,22 @@ def eft_error_handling(self, row, error_msg: str, capture_error: bool = True, ta
             send_error_email(email_service_params)
 
 
+def _validate_configuration(context: EFTReconciliation) -> bool:
+    """Validate required configuration is available."""
+    is_valid = True
+    if current_app.config.get("EFT_TDI17_LOCATION_ID") is None:
```
Comment (seeker25) [inline pay-queue/src/pay_queue/services/eft/eft_reconciliation.py:67]: hehe MAKE IT BLOW UP

**Review by seeker25 — APPROVED** [review]
LGTM

File: pay-queue/devops/vaults.gcp.env [hunk 6]
```diff
@@ -31,3 +31,6 @@ NOTIFY_API_URL="op://API/$APP_ENV/notify-api/NOTIFY_API_URL"
 NOTIFY_API_VERSION="op://API/$APP_ENV/notify-api/NOTIFY_API_VERSION"
 JWT_OIDC_ISSUER="op://keycloak/$APP_ENV/jwt-base/JWT_OIDC_ISSUER"
 PAY_CONNECTOR_AUTH="op://relationship/$APP_ENV/pay-api/PAY_CONNECTOR_AUTH"
+EFT_TDI17_LOCATION_ID="op://relationship/$APP_ENV/pay-queue/EFT_TDI17_LOCATION_ID"
```
[the remark this answers is not in the document]
Comment (ochiu) [reply pay-queue/devops/vaults.gcp.env:34]: Yes sir

**Pull request merged**